servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -205,12 +215,38 @@ public final HttpServerBuilder appendConnectionAcceptorFilter(final ConnectionAc
      * @param factory {@link StreamingHttpServiceFilterFactory} to append.
      * @return {@code this}
      */
-    public final HttpServerBuilder appendServiceFilter(final StreamingHttpServiceFilterFactory factory) {
+    public final HttpServerBuilder appendNonOffloadingServiceFilter(final StreamingHttpServiceFilterFactory factory) {
         requireNonNull(factory);
-        serviceFilter = appendFilter(serviceFilter, factory);
-        if (!influencerChainBuilder.appendIfInfluencer(factory)) {
-            influencerChainBuilder.append(defaultStreamingInfluencer());
+        HttpExecutionStrategy factoryStrategy = influenceStrategy(factory, OFFLOAD_NONE_STRATEGY);
```

Comment:
I agree that requiring the user to implement `HttpExecutionStrategyInfluencer` might just cause them to write a GIGO implementation without being honest. I added the checks in append to help avoid honest mistakes, but it is not possible to prevent intentional misuse.

Reply:
Agreed, users can incorrectly implement the influencer method. Or influencer may not be updated after the filter code changes. However, it's a bigger problem that will also impact `appendServiceFilter` with API conversions. 
For now, we can trust that users implemented the contract correctly and require filters to implement influencer for `appendNonOffloadingServiceFilter`